### PR TITLE
Update snapshotter images to v5.0.0-rc1

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <ggriffiths@purestorage.com>

**What type of PR is this?**
Updates snapshotter images to v5.0.0-rc1

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by ggriffiths** [comment]
cc @xing-yang

File: deploy/kubernetes-1.20-test/hostpath/csi-hostpath-snapshotter.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ spec:
       serviceAccountName: csi-snapshotter
       containers:
         - name: csi-snapshotter
-          image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.1
+          image: gcr.io/k8s-staging-sig-storage/csi-snapshotter:v5.0.0-rc1
```
Comment (xing-yang) [inline deploy/kubernetes-1.20-test/hostpath/csi-hostpath-snapshotter.yaml:40]: Can you add a TODO here to replace this with the released image?  Same comment for other places.

**Comment by xing-yang** [comment]
Once https://github.com/kubernetes-csi/external-snapshotter/pull/609 is merged, let's see if this can pass the canary test, i.e., ci-kubernetes-csi-canary-on-kubernetes-1-20

File: deploy/kubernetes-1.20-test/hostpath/csi-hostpath-snapshotter.yaml (hunk 1 above)
Comment (pohly) [reply deploy/kubernetes-1.20-test/hostpath/csi-hostpath-snapshotter.yaml:40]: Even better, open an issue that tracks all remaining work.

No-one checks TODOs in code...

Comment (ggriffiths) [reply deploy/kubernetes-1.20-test/hostpath/csi-hostpath-snapshotter.yaml:42]: Added a TODO and issue:
https://github.com/kubernetes-csi/external-snapshotter/issues/610

**Pull request merged**